plugins/out_datadog/datadog_conf.c

```diff
@@ -98,6 +99,20 @@ struct flb_out_datadog *flb_datadog_conf_create(struct flb_output_instance *ins,
         return NULL;
     }
 
+    /* Parse site parameter early so it's available for host construction */
+    tmp = flb_output_get_property("site", ins);
```

Comment:
if the config map is adjusted, here you will only need to check if ctx->site is not null and you avoid the allocations set